quarkchain/evm/messages.py

```diff
@@ -407,102 +406,95 @@ def apply_transaction(state, tx: transactions.Transaction, tx_wrapper_hash):
     ext = VMExt(state, tx.sender, gasprice)
 
     contract_address = b""
-    if tx.is_cross_shard:
-        local_gas_used = intrinsic_gas
-        remote_gas_reserved = 0
-        if transfer_failure_by_posw_balance_check(ext, message):
-            success = 0
-            # Currently, burn all gas
-            local_gas_used = tx.startgas
-        elif tx.to == b"":
-            state.delta_token_balance(tx.sender, tx.transfer_token_id, -tx.value)
-            remote_gas_reserved = tx.startgas - intrinsic_gas
-            ext.add_cross_shard_transaction_deposit(
-                quarkchain.core.CrossShardTransactionDeposit(
-                    tx_hash=tx_wrapper_hash,
-                    from_address=quarkchain.core.Address(
-                        tx.sender, tx.from_full_shard_key
-                    ),
-                    to_address=quarkchain.core.Address(
-                        mk_contract_address(
-                            tx.sender,
-                            state.get_nonce(tx.sender),
-                            tx.from_full_shard_key,
-                        ),
-                        tx.to_full_shard_key,
-                    ),
-                    value=tx.value,
-                    # convert to genesis token and use converted gas price
-                    gas_token_id=state.genesis_token,
-                    gas_price=gasprice,
-                    transfer_token_id=tx.transfer_token_id,
-                    message_data=tx.data,
-                    create_contract=True,
-                    gas_remained=remote_gas_reserved,
-                    refund_rate=refund_rate,
-                )
-            )
-            success = 1
-        else:
-            state.delta_token_balance(tx.sender, tx.transfer_token_id, -tx.value)
-            if (
-                state.qkc_config.ENABLE_EVM_TIMESTAMP is None
-                or state.timestamp >= state.qkc_config.ENABLE_EVM_TIMESTAMP
-            ):
-                remote_gas_reserved = tx.startgas - intrinsic_gas
-            ext.add_cross_shard_transaction_deposit(
-                quarkchain.core.CrossShardTransactionDeposit(
-                    tx_hash=tx_wrapper_hash,
-                    from_address=quarkchain.core.Address(
-                        tx.sender, tx.from_full_shard_key
+    if not tx.is_cross_shard:
```

Comment:
changed the order of control flow for better readability. no actual change to the logic